src/components/coin-row/TransactionCoinRow.js

```diff
@@ -116,14 +118,19 @@ export default function TransactionCoinRow({ item, ...props }) {
     }
 
     if (hash) {
-      let buttons = ['View on Etherscan', ...(ios ? ['Cancel'] : [])];
+      let buttons = [
+        ...(canBeResubmitted ? ['Speed Up', 'Attempt Cancellation'] : []),
+        'View on Etherscan',
+        ...(ios ? ['Cancel'] : []),
+      ];
       if (showContactInfo) {
         buttons.unshift(contact ? 'View Contact' : 'Add to Contacts');
       }
 
       showActionSheetWithOptions(
         {
-          cancelButtonIndex: showContactInfo ? 2 : 1,
+          cancelButtonIndex:
+            (canBeResubmitted ? 2 : 0) + (showContactInfo ? 2 : 1),
```

Comment:
If `canBeResubmitted` and `showContactInfo` are both true, the index would add up to 4, which doesn't exist. Given that the Cancel button is always the last in the list, perhaps we should just use `buttons.length - 1` as the `cancelButtonIndex`

Reply:
That would be more elegant, yes. But index 4 exists:

[
"Speed up", 
"Attempt Cancellation", 
"View on Etherscan", 
"Add contact",
"Cancel" 
]

Reply:
Ah gotcha!